tests/course/test_for_course.py

```diff
@@ -0,0 +1,185 @@
+        assert teacherCourses1 in TeacherCourses.objects.in_price_range(PRICE, PRICE)
+
+    def test_category_filter(self, teacherCourses1):
+        assert teacherCourses1 in TeacherCourses.objects.in_category(CATEGORY)
+        assert teacherCourses1 not in TeacherCourses.objects.in_category(CATEGORY + "not a category")
+
+    def test_experience_filter(self, teacherCourses1):
+        assert teacherCourses1 in TeacherCourses.objects.got_experience(YEARS_OF_EXP)
+        assert teacherCourses1 in TeacherCourses.objects.got_experience(YEARS_OF_EXP - 1)
+        assert teacherCourses1 not in TeacherCourses.objects.got_experience(YEARS_OF_EXP + 1)
+
+    def test_search_name_filter(self, teacherCourses1):
+        assert teacherCourses1 in TeacherCourses.objects.search_name(COURSE_NAME)
+        assert teacherCourses1 in TeacherCourses.objects.search_name(DESCRIPTION)
+        assert teacherCourses1 in TeacherCourses.objects.search_name(COURSE_NAME[:2])
+        assert teacherCourses1 in TeacherCourses.objects.search_name(DESCRIPTION[:2])
+        assert teacherCourses1 not in TeacherCourses.objects.search_name("not a course name")
+        assert teacherCourses1 not in TeacherCourses.objects.search_name("not a description")
+
+    def test_difficulty_level_filter(self, teacherCourses1):
+        assert teacherCourses1 in TeacherCourses.objects.get_level(DIFFICULTY)
+        assert teacherCourses1 not in TeacherCourses.objects.get_level(DIFFICULTY + "not a difficulty level")
+        assert teacherCourses1 not in TeacherCourses.objects.get_level('B')
+
+    def test_get_teacher_courses_filter(self, teacherCourses1, teacherCourses2):
+        assert teacherCourses1 in TeacherCourses.objects.get_teacher_courses(teacherCourses1.teacher_id.pk)
+        assert teacherCourses1 not in TeacherCourses.objects.get_teacher_courses(teacherCourses2.teacher_id.pk)
+        assert teacherCourses2 not in TeacherCourses.objects.get_teacher_courses(teacherCourses1.teacher_id.pk)
+        assert teacherCourses2 in TeacherCourses.objects.get_teacher_courses(teacherCourses2.teacher_id.pk)
+
+    # CHECK INVALED DATA INPUT
+
+    def test_course_name(self, db, user1):
+        # TOO LONG
+        with pytest.raises(ValidationError):
+            teacher_course = TeacherCourses(teacher_id=user1, course_name=NAME_TOO_LONG, description=DESCRIPTION,
+                                            difficulty_level=DIFFICULTY, category=CATEGORY,
+                                            price=PRICE, years_of_experience=YEARS_OF_EXP)
+            teacher_course.save()
+        # EMPTY
+        with pytest.raises(ValidationError):
+            teacher_course = TeacherCourses(teacher_id=user1, description=DESCRIPTION,
+                                            difficulty_level=DIFFICULTY, category=CATEGORY, price=PRICE,
+                                            years_of_experience=YEARS_OF_EXP)
+            teacher_course.save()
+
```

Comment:
instead of creating a new teacher course maybe you can take an existing fixture and change the params required for the tests, it will make it more readable and focused in my opinion.